.github/workflows/build-and-publish.yml

```
@@ -2,10 +2,12 @@ name: Build and Publish
 
 on:
   push:
+    branches: [ 'main' ]  # Trigger on push to 'main'
+    tags: [ '[0-9]+.[0-9]+.[0-9]' ]  # Trigger on version tags like '0.1.0'
```

Without this final `+`, a version like 1.2.12 wouldn't pass.
```suggestion
    tags: [ '[0-9]+.[0-9]+.[0-9]+' ]  # Trigger on version tags like '0.1.0'
```